analyzers/src/SonarAnalyzer.Common/Helpers/EnumerableExtensions.cs

```diff
@@ -149,5 +149,24 @@ public static IEnumerable<T> WhereNotNull<T>(this IEnumerable<Nullable<T>> enume
         /// </summary>
         public static string JoinNonEmpty(this IEnumerable<string> enumerable, string separator) =>
             string.Join(separator, enumerable.Where(x => !string.IsNullOrWhiteSpace(x)));
+
+        /// <summary>
+        /// Concatenates the members of <paramref name="values"/> using "and" as the last separator and using a
+        /// <see href="https://en.wikipedia.org/wiki/Serial_comma">serial comma</see>.
+        /// <list type="table">
+        /// <item><c>[a, b, c] => "a, b, and c"</c></item>
+        /// <item><c>[a, b] => "a and b"</c></item>
+        /// <item><c>[a] => "a"</c></item>
+        /// <item><c>[] or null => ""</c></item>
+        /// </list>
+        /// </summary>
+        public static string JoinAnd<T>(this IEnumerable<T> values) =>
```

Comment:
I think this is a useful method to have for error messages (e.g. [here](https://github.com/SonarSource/sonar-dotnet/blob/d811b627d0effd296803fe64b28b3e955613c3b4/analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs#L144)). I don't use it in this PR anymore, but if we want to keep it, I will create a new PR for it and add some tests.